main/lsp/LSPConfiguration.cc

```diff
@@ -84,6 +84,17 @@ LSPClientConfiguration::LSPClientConfiguration(const InitializeParams &params) {
                 clientHoverMarkupKind = getPreferredMarkupKind(contentFormat);
             }
         }
+        if (textDocument->codeAction) {
+            auto &codeAction = *textDocument->codeAction;
+            if (codeAction->dataSupport) {
+                clientCodeActionDataSupport = codeAction->dataSupport.value_or(false);
+            }
+
+            if (codeAction->resolveSupport) {
```

Comment:
(This is non-blocking)

`std::optional` coercion to `bool` is neat, but I think the explicit `.has_value()` is much clearer, especially when the optional is holding a value that can be coerced to a `bool` itself (pointers, integers, `bool`, etc.).  I don't think we have agreement on this yet, though, and we obviously don't have any linting for it or such, so this is mostly food for thought.